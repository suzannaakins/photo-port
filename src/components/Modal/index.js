import React from 'react';

const Modal = ({ onClose, currentPhoto }) => {
    const { name, category, description, index } = currentPhoto;

    return (
        <div className="modalBackdrop">
            <div className="modalContainer">
                <button onClick={onClose} type="button">
                    X
              </button>
                <h3 className="modalTitle">{name}</h3>
                <img src={require(`../../assets/large/${category}/${index}.jpg`).default} alt="current category" />
                <p>
                    {description}
                </p>
            </div>
        </div>
    );
}

export default Modal; 