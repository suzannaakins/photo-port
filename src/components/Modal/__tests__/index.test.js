import React from 'react';
import { render, cleanup, fireEvent, getNodeText, getByTestId } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();

const currentPhoto = {
    name: 'Grocery aisle',
    category: 'commercial',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    index: 1
}

afterEach(cleanup);

describe('Modal component', () => {
    it('renders', () => {
        render(<Modal
            currentPhoto={currentPhoto}
            toggleModal={mockToggleModal} />)
    })
    //snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Modal
            currentPhoto={currentPhoto}
            toggleModal={mockToggleModal} />);
        //assert value comparison
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('Click Event', () => {
    it('calls onClose handler', () => {

        //Arrange: render modal
        const { getByText } = render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
        />);

        //Act: simuulate the click event
        fireEvent.click(getByText('X'));

        //Assert: expected matcher
        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    })
})