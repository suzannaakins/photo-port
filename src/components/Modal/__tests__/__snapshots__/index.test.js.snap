// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <button
        type="button"
      >
        X
      </button>
      <h3
        class="modalTitle"
      >
        Grocery aisle
      </h3>
      <img
        alt="current category"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
    </div>
  </div>
</DocumentFragment>
`;
